src/main/java/com/codeus/winter/config/impl/BeanDefinitionRegistryImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.codeus.winter.config.impl;
+
+import com.codeus.winter.config.BeanDefinition;
+import com.codeus.winter.config.BeanDefinitionRegistry;
+import com.codeus.winter.exception.BeanDefinitionStoreException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class BeanDefinitionRegistryImpl implements BeanDefinitionRegistry {
+    private final Map<String, BeanDefinition> beanDefinitionMap = new HashMap<>();
+
+    @Override
+    public void registerBeanDefinition(String beanName, BeanDefinition beanDefinition) {
+        if (beanDefinitionMap.containsKey(beanName)) {
+            throw new BeanDefinitionStoreException("Cannot register bean definition with name '" + beanName +
+                    "': another bean with the same name already exists and overriding is not allowed.");
+        }
+        beanDefinitionMap.put(beanName, beanDefinition);
+    }
+
+    @Override
+    public void removeBeanDefinition(String beanName) {
+        if (!beanDefinitionMap.containsKey(beanName)) {
+            throw new BeanDefinitionStoreException("No bean definition found for name '" + beanName + "'");
```

Comment:
pls use `String.format`

Reply:
If you only have one parameter to substitute, using + is simpler and more intuitive.